crates/astria-sequencer/src/checked_actions/fee_asset_change.rs

```diff
@@ -66,6 +67,48 @@ impl CheckedFeeAssetChange {
             &sudo_address == self.tx_signer.as_bytes(),
             "transaction signer not authorized to change fee assets"
         );
+
+        // NOTE: To allow `app_legacy_execute_transactions_with_every_action_snapshot` to continue
+        // to pass, we need to make an exception for `test-0` here to allow the test tx to be
+        // constructed. This exception can be removed once the legacy test is removed.
+        #[cfg(test)]
```

Comment:
what is the legacy test for? is it to ensure backwards-compatibility? can we change the test to have `test-0` be an existing asset (which will change the snapshot but can add an explanation of why and that it's not breaking)

Reply:
It was really to help ensure backwards-compatibility during the large changeset in #2142.  Now that that PR has merged, it's safe to replace that legacy test with something more convenient, and which will let us remove these workarounds altogether.

This is ticketed in #2157.